### PR TITLE
Optimize store profiler: remove feature flag

### LINKED ISSUE #10492: Remove feature flag and release new profiler flow

### PR DESCRIPTION
**Pull request by itsmeichigo**
Closes #10492 

## Description

We have previously removed the checks for the feature flag `optimizeProfilerQuestions`, so this PR removes it in the code.

## Testing instructions

CI passing should be enough.

## Screenshots

N/A

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -3,7 +3,7 @@
 15.0
 -----
 - [*] The store name can now be updated from the Settings screen. [https://github.com/woocommerce/woocommerce-ios/pull/10485]
-
+- [**] The store creation flow has been optimized to start store creation immediately and show profiler questions afterward. [https://github.com/woocommerce/woocommerce-ios/issues/10374]
```
Comment (selanthiraiyan) [inline RELEASE-NOTES.txt]: Should we replace the issue link with PR link/s instead?

**Review by selanthiraiyan — APPROVED** [review]
:shipit:

File: RELEASE-NOTES.txt (hunk 1 above)
Comment (itsmeichigo) [reply RELEASE-NOTES.txt]: There are so many links, I don't think one would be enough. Please feel free to suggest the links you think is suitable for this.

Comment (selanthiraiyan) [reply RELEASE-NOTES.txt]: I agree. Maybe https://github.com/woocommerce/woocommerce-ios/pull/10473, https://github.com/woocommerce/woocommerce-ios/pull/10466 and https://github.com/woocommerce/woocommerce-ios/pull/10485.

Please also feel free to ignore my suggestion and keep the original issues link.

Comment (itsmeichigo) [reply RELEASE-NOTES.txt]: I added an item for the store name item in Settings screen already so I'll just add the other two links then. Thanks!

**Pull request merged**